pkg/operator2/route.go

```diff
@@ -36,14 +43,13 @@ func (c *authOperator) handleRoute(ingress *configv1.Ingress) (*routev1.Route, *
 	route.Spec.Host = host
 
 	if err := isValidRoute(route, ingress); err != nil {
-		// TODO remove this delete so that we do not lose the early creation timestamp of our route
-		// delete the route so that it is replaced with the proper one in next reconcile loop
-		klog.Infof("deleting invalid route: %#v", route)
-		opts := &metav1.DeleteOptions{Preconditions: &metav1.Preconditions{UID: &route.UID}}
-		if err := c.route.Delete(route.Name, opts); err != nil && !errors.IsNotFound(err) {
-			klog.Infof("failed to delete invalid route: %v", err)
+		klog.Infof("route invalid: %v, restoring", err)
+		routeUpdate := defaultRoute(ingress)
+		routeUpdate.ObjectMeta = route.ObjectMeta // Preserve metadata
```

Comment:
I went with partial preservel in https://github.com/mrogers950/cluster-authentication-operator/pull/2